app/bundles/LeadBundle/Tests/Controller/AjaxControllerFunctionalTest.php

```diff
@@ -514,6 +514,77 @@ public function testContactListActionSuggestionsByNonAdminUser(): void
         self::assertSame('User 4', $foundNames[1]);
     }
 
+    public function testUpdateLeadFieldOrderChoiceListAction(): void
+    {
+        $tests = [
+            [
+                'object'          => 'lead',
+                'group'           => 'core',
+                'expectedOptions' => ['Fax', 'Website'],
+            ],
+            [
+                'object'          => 'lead',
+                'group'           => 'social',
+                'expectedOptions' => ['Facebook', 'Foursquare', 'Instagram'],
+            ],
+            [
+                'object'          => 'company',
+                'group'           => 'core',
+                'expectedOptions' => [],
+            ],
+        ];
+
+        foreach ($tests as $test) {
+            $object          = $test['object'];
+            $group           = $test['group'];
+            $expectedOptions = $test['expectedOptions'];
+
+            $payload = [
+                'action' => 'lead:updateLeadFieldOrderChoiceList',
+                'object' => $object,
+                'group'  => $group,
+            ];
+
+            $this->client->request(
+                Request::METHOD_POST,
+                '/s/ajax',
+                $payload,
+                [],
+                $this->createAjaxHeaders()
+            );
+
+            // Get the response HTML
+            $response    = $this->client->getResponse();
+            $htmlContent = $response->getContent();
+
+            // Assert the response is successful
+            $this->assertTrue($response->isOk(), "Response was not OK for object: $object, group: $group");
+            $this->assertStringNotContainsString('<form', $htmlContent, 'Response contains a form instead of just field order.');
+            $this->assertStringContainsString('<select', $htmlContent, 'Response contains select tag.');
+
+            // Parse the HTML content using DOMDocument
+            $dom = new \DOMDocument();
+            $dom->loadHTML($htmlContent);
+            $select  = $dom->getElementsByTagName('select')->item(0);
+            $options = $select->getElementsByTagName('option');
+
+            $actualOptions = [];
+            foreach ($options as $option) {
+                if ($option->textContent) {
+                    // Get the text content of each <option>
+                    $actualOptions[] = trim($option->textContent);
+                }
+            }
+            // Assert that the actual options match the expected options
+            if (empty($expectedOptions)) {
+                $this->assertEmpty($actualOptions);
+            }
+            foreach ($expectedOptions as $expectedValue) {
+                $this->assertContains($expectedValue, $actualOptions, "Missing expected option '$expectedValue' for object: $object, group: $group");
+            }
+        }
+    }
```

Comment:
This could be simplified with PHPUNIT's data providers. What do you think?
```suggestion
    /**
     * @dataProvider leadFieldOrderChoiceListProvider
     *
     * @param string[] $expectedOptions
     */
    public function testUpdateLeadFieldOrderChoiceListAction(string $object, string $group, array $expectedOptions): void
    {
        $payload = [
            'action' => 'lead:updateLeadFieldOrderChoiceList',
            'object' => $object,
            'group'  => $group,
        ];

        $this->client->request(
            Request::METHOD_POST,
            '/s/ajax',
            $payload,
            [],
            $this->createAjaxHeaders()
        );

        // Get the response HTML
        $response    = $this->client->getResponse();
        $htmlContent = $response->getContent();

        // Assert the response is successful
        $this->assertTrue($response->isOk(), "Response was not OK for object: $object, group: $group");
        $this->assertStringNotContainsString('<form', $htmlContent, 'Response contains a form instead of just field order.');
        $this->assertStringContainsString('<select', $htmlContent, 'Response contains select tag.');

        // Parse the HTML content using DOMDocument
        $dom = new \DOMDocument();
        @$dom->loadHTML($htmlContent);
        $select  = $dom->getElementsByTagName('select')->item(0);
        $options = $select->getElementsByTagName('option');

        $actualOptions = [];
        foreach ($options as $option) {
            if ($option->textContent) {
                // Get the text content of each <option>
                $actualOptions[] = trim($option->textContent);
            }
        }
        // Assert that the actual options match the expected options
        if (empty($expectedOptions)) {
            $this->assertEmpty($actualOptions);
        }
        foreach ($expectedOptions as $expectedValue) {
            $this->assertContains($expectedValue, $actualOptions, "Missing expected option '$expectedValue' for object: $object, group: $group");
        }
    }

    public function leadFieldOrderChoiceListProvider(): \Generator
    {
        yield ['lead', 'core', ['Fax', 'Website']];
        yield ['lead', 'social', ['Facebook', 'Foursquare', 'Instagram']];
        yield ['company', 'core', []];
    }
```

Reply:
Yes it definitely simplifies it, didn't realise I could do that, thanks!